components/Users/UserModal/userModal.tsx

```diff
@@ -23,9 +26,69 @@ const UserModal: React.FunctionComponent<UserModalProps> = ({
   setIsOpen,
   isFamilyPath,
   setIsOpenFam,
+  setEdited,
+  refresh,
 }: UserModalProps) => {
+  const [isEditing, setIsEditing] = useState(false);
+  const [currUser, setCurrUser] = useState(user);
+  const [role, setRole] = useState(user?.role);
+  const [email, setEmail] = useState(user?.email);
+  const [error, setError] = useState("");
+  const [phoneNumber, setPhoneNumber] = useState(user?.phone_number);
+
+  useEffect(() => {
+    const callFunc = async () => {
+      if (user != undefined) {
+        const data = await getUser(user?.user_id);
+        setCurrUser(data);
+      }
+    };
+    callFunc();
+  }, []);
+
+  async function onSubmit(event?: React.BaseSyntheticEvent): Promise<void> {
+    event?.preventDefault();
+    const newData = {};
+    try {
+      if (role && role != user?.role) {
+        if (role !== "Head" && role !== "Parent" && role !== "Child") {
+          throw new Error(
+            "Input for role must be 'Head' or 'Parent' or 'Child'"
```

Comment:
https://user-images.githubusercontent.com/28970991/142943023-3865277f-941a-4dd7-869d-0ce10bd51e15.mov

just one last detail, do you think it would be possible to add a loading circle/something to indicate that the user's data is being updated?

Reply:
@hukellyy what do you think?